### ISSUE TITLE
Add variant query documentation

### ISSUE THREAD

**Issue by mbaudis** [ga4gh-beacon/beacon-v2#5]
There s now a placeholder page for genomic query documentation. Please add content (e.g. transfer from the query scouts document) [here](https://github.com/ga4gh-beacon/beacon-v2-unity-testing/blob/main/docs/variant-queries.md). (also @babisingh).

**Comment by redmitry** [comment]
(how) does this documentations align with the Beacon v2 spec.?
In the [variant-queries.md](https://github.com/ga4gh-beacon/beacon-v2/blob/main/docs/variant-queries.md)
https://docs.genomebeacons.org/schemas-md/genomicVariations_defaultSchema/

~~~
?referenceName=NC_000017.11&start=7577120&referenceBases=G&alternateBases=A
~~~
On the other hand, [/genomicVariations/endpoints.json](https://github.com/ga4gh-beacon/beacon-v2-Models/blob/main/BEACON-V2-Model/genomicVariations/endpoints.json) defines:
~~~json
"paths": {
    "/g_variants": {
      "get": {
        "parameters": [
          { "$ref": "#/components/parameters/requestedSchema" },
          { "$ref": "#/components/parameters/skip" },
          { "$ref": "#/components/parameters/limit" },
          { "$ref": "#/components/parameters/includeResultsetResponses" },
          { "$ref": "#/components/parameters/genomicAlleleShortForm" },
          { "$ref": "#/components/parameters/geneId" },
          { "$ref": "#/components/parameters/aminoacidChange" }
        ]
~~~

Cheers,

Dmitry

**Comment by mbaudis** [comment]
Well, ‘endpoints.json` is wrong since it should use `requestParameters.json`.

**Comment by jrambla** [comment]
I don't think that would work Michael, OpenAPI has its own syntax (that follows their schema).
Additionally GET and POST parameters are (partially) different and I guess that what Dmitry is pointing at is a discrepancy between the parameters in the GET OpenAPI file and the ones used in the examples.
None of the parameters in the example query are defined in the OpenAPI GET parameters list.
NOTE: the parameters between GET and POST _DOESN'T NEED TO MATCH_ as they are used in different context.
In the design of the GET parameters, I've prioritized compactness vs. one-to-one equivalence with POST.

In summary, the commit is not correct, and we either update the examples or change the OpenAPI to match the POST.

**Comment by redmitry** [comment]
>NOTE: the parameters between GET and POST DOESN'T NEED TO MATCH as they are used in different context.
Agree, but I am no qualified to make a decision about GET params.
Anyway happy to throw the ball ;-).

**Comment by mbaudis** [comment]
Well, I had a different issue about the need to adjust/document the GET parameters somewhere …

Obviously the examples are correct, mimicking existing practices & the POST parameters.

So: could someone please add the parameters to the endpoints? Or get rid of the OpenAPI stuff & just document the translation of POST to GET parameters (which AFAIK will be the one most people will use). The OpenAPI… question has come up before - I personally implement by schema & documentation, and this is a very confusing remnant with apparently frequent collisions if not kept in sync.

**Comment by redmitry** [comment]
OK. I'll do the PR and then Jordi will review.

**Comment by mbaudis** [comment]
… and definitely it doesn’t make sense to have skip or geneId but not the standard variant parameters.

**Comment by jrambla** [comment]
I disagree with you on that Michael,  having lots of parameters in GET makes it unnecessarily complex, IMO. But let's see Dmitry PR and continue from there.
We have the request (indeed from the PRC if my memory is not tricking me) to have an OpenAPI (or alike) version to "visualize/interact/play with/understand" the API itself. 
So far the OpenAPI has been/is the only candidate for that.
I'm not a fan either, as it adds another language to the playfield and eventually more noise, but... do we have an alternative?

P.D.: The examples are only as much correct as much as they follow the spec ;-)

**Comment by mbaudis** [comment]
@jrambla But not having the - constantly used, documented, v1 basic… - parameters on GET makes it a fail. I would kill the OpenAPI stuff & document the syntax for GET parameter use, for a “reasonable subset”. We’re running Progenetix completely on GET, while also supporting POST, with minimal liberties taken and some generalizations (i.e. includeDescendingTerms in filters is a global parameter, not per filter).

But absolutely - the essential query parameters for genomicVariations … documented have never been considered non-standard GET parameters.

**Comment by mbaudis** [comment]
… so the primary access to Beacon resources is through a set of well defined query parameters, readable as URLs, with some more fancy stuff (actually only some filter options? Also Gargantuan parameter lists etc.) needing an object/POST syntax. And if OpenAPI cannot make use of documents prototyping such parameters (as in “just point to the variant parameters doc for all of the endpoints”) - well… Have fun writing many lines, repeatedly.

**Comment by mbaudis** [comment]
> The examples are only as much correct as much as they follow the spec ;-)

The spec is incomplete.

**Comment by mbaudis** [comment]
Closing this since parameter now in endpoints.
